src/coreComponents/common/TimingMacros.hpp

```diff
@@ -45,40 +39,107 @@ namespace timingHelpers
   }
 }
 
+
+#if defined( GEOSX_USE_CUDA_NVTOOLSEXT )
+
+#include "nvToolsExt.h"
+
+namespace timingHelpers
+{
+  enum NVTXColors {
+    GREEN = 0xff00ff00,
+    BLUE = 0xff0000ff,
+    YELLOW = 0xffffff00,
+    PURPLE = 0xffff00ff,
+    AQUA = 0xff00ffff,
+    RED = 0xffff0000,
+    WHITE = 0xffffffff
+  };
+
+  class NVTXScopeTracer {
+  public:
+    NVTXScopeTracer(const char* name, NVTXColors colorID) {
+      nvtxEventAttributes_t eventAttrib = { 0 };
+      eventAttrib.version = NVTX_VERSION;
+      eventAttrib.size = NVTX_EVENT_ATTRIB_STRUCT_SIZE;
+      eventAttrib.colorType = NVTX_COLOR_ARGB;
+      eventAttrib.color = colorID;
+      eventAttrib.messageType = NVTX_MESSAGE_TYPE_ASCII;
+      eventAttrib.message.ascii = name;
+      nvtxRangePushEx(&eventAttrib);
+    }
+    ~NVTXScopeTracer() {
+      nvtxRangePop();
+    }
+  };
+
+}
+/// Mark a scope with NVTX with name and color (only a local helper should not be used elsewhere)
+#  define GEOSX_NVTX_MARK_SCOPE_2(name, color) timingHelpers::NVTXScopeTracer __FILE__ ## _ ## __LINE__ ## _ ## scopeTracer = timingHelpers::NVTXScopeTracer(name, color)
```

Comment:
```suggestion
#  define GEOSX_NVTX_MARK_SCOPE_COLORED(name, color) timingHelpers::NVTXScopeTracer __FILE__ ## _ ## __LINE__ ## _ ## scopeTracer = timingHelpers::NVTXScopeTracer(name, color)```

Reply:
There are triple backquotes unwanted at the end + have to change the call also. I'll do it manually.